controllers/vaultstaticsecret_controller.go

```diff
@@ -79,54 +78,24 @@ func (r *VaultStaticSecretReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		requeueAfter = computeHorizonWithJitter(d)
 	}
 
-	var resp *api.KVSecret
-	var respErr error
-	switch o.Spec.Type {
-	case consts.KVSecretTypeV1:
-		if w, err := c.KVv1(o.Spec.Mount); err != nil {
-			return ctrl.Result{}, err
-		} else {
-			resp, respErr = w.Get(ctx, o.Spec.Path)
-		}
-	case consts.KVSecretTypeV2:
-		if w, err := c.KVv2(o.Spec.Mount); err != nil {
-			return ctrl.Result{}, err
-		} else {
-			if o.Spec.Version == 0 {
-				resp, respErr = w.Get(ctx, o.Spec.Path)
-			} else {
-				resp, respErr = w.GetVersion(ctx, o.Spec.Path, o.Spec.Version)
-			}
-		}
-	default:
-		err := fmt.Errorf("unsupported secret type %q", o.Spec.Type)
-		logger.Error(err, "")
+	kvReq, err := newKVRequest(o.Spec)
+	if err != nil {
 		r.Recorder.Event(o, corev1.EventTypeWarning, consts.ReasonVaultStaticSecret, err.Error())
 		return ctrl.Result{}, err
 	}
 
-	if respErr != nil {
-		logger.Error(err, "Failed to read Vault secret")
+	resp, err := c.Read(ctx, kvReq)
+	if err != nil {
 		r.Recorder.Eventf(o, corev1.EventTypeWarning, consts.ReasonVaultClientError,
 			"Failed to read Vault secret: %s", err)
-		return ctrl.Result{}, nil
+		return ctrl.Result{RequeueAfter: requeueAfter}, nil
 	}
 
-	if resp == nil {
-		logger.Error(nil, "empty Vault secret", "mount", o.Spec.Mount, "path", o.Spec.Path)
-		r.Recorder.Eventf(o, corev1.EventTypeWarning, consts.ReasonVaultClientError,
-			"Vault secret was empty, mount %s, path %s", o.Spec.Mount, o.Spec.Path)
-		return ctrl.Result{
-			RequeueAfter: requeueAfter,
-		}, nil
-	}
-
-	data, err := makeK8sSecret(resp)
+	data, err := vault.MarshalData(resp.Data(), resp.Secret().Data)
 	if err != nil {
-		logger.Error(err, "Failed to construct k8s secret")
 		r.Recorder.Eventf(o, corev1.EventTypeWarning, consts.ReasonVaultClientError,
-			"Failed to construct k8s secret: %s", err)
-		return ctrl.Result{}, err
+			"Invalid Vault Secret data: %s", err)
+		return ctrl.Result{RequeueAfter: requeueAfter}, nil
```

Comment:
Similar question here.

Reply:
Addressed in 7fa55d5ffeda41e8c2705cc32de3385a04e0ce1e